### PR TITLE
fixes some bugs and add tests to improve code coveragwe

### PR DESCRIPTION
**Pull request by mssfang**
- Add client constants class to maintains all cosntants for App configuration client.
- 'id' is type of String, could be null or empty

- Added tests for code coverage for `ConfigurationClientBuilder` and ``SettingSelector.

### CODE REVIEW

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClientBuilder.java [hunk 1]
```diff
@@ -355,14 +355,14 @@ public ConfigurationClientBuilder configuration(Configuration configuration) {
...
-     * Sets the {@link HttpPipelinePolicy} that is used when each request is sent.
+     * Sets the {@link RetryPolicy} that is used when each request is sent.
      *
      * The default retry policy will be used if not provided {@link ConfigurationClientBuilder#buildAsyncClient()}
      * to build {@link ConfigurationAsyncClient} or {@link ConfigurationClient}.
      * @param retryPolicy user's retry policy applied to each request.
      * @return The updated ConfigurationClientBuilder object.
      */
-    public ConfigurationClientBuilder retryPolicy(HttpPipelinePolicy retryPolicy) {
+    public ConfigurationClientBuilder retryPolicy(RetryPolicy retryPolicy) {
```
Comment (alzimmermsft) [inline sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClientBuilder.java]: This breaks public API and won't be allowed. The old method either needs to be deprecated and this put in it's place or remmain as is.

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/implementation/ConfigurationClientCredentials.java [hunk 2]
```diff
@@ -212,7 +211,7 @@ String id() {
             this.id = id;
             this.secret = secret;
 
-            if (this.baseUri == null || this.id == null || this.secret == null) {
+            if (this.baseUri == null || CoreUtils.isNullOrEmpty(this.id) || this.secret == null) {
```
Comment (alzimmermsft) [inline sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/implementation/ConfigurationClientCredentials.java]: Why is id treated differently than baseUri or secret, won't all have issues if null or empty?

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClientBuilder.java (hunk 1 above)
Comment (mssfang) [reply sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClientBuilder.java]: ok. I will revert back the change

File: sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/implementation/ConfigurationClientCredentials.java (hunk 2 above)
Comment (mssfang) [reply sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/implementation/ConfigurationClientCredentials.java]: id' is type of String, could be null or empty
but baseUri is URL type, `private final URL baseUri;`.

**Pull request merged**